### ISSUE TITLE
New components : list box and hierarchal list

### ISSUE THREAD

**Issue by Sr44102** [dxc-technology/halstack-react#1405]
**Is your feature request related to a problem? Please describe.**
We are from Assure Reinsurance team. While developing a Web application for existing desktop application we came across few components which are not available in CDK toolkit. 
1. List box with multi selector
2. Hierarchical list 

**Existing designs in desktop application**
List box
![image](https://user-images.githubusercontent.com/119670575/205294510-5df76c09-8c19-49d0-a6f6-ffe3f503844b.png)

Hierarchical list and table
![image](https://user-images.githubusercontent.com/119670575/205294567-9436dc8c-2510-4cf1-84e3-a6da29d8a9ca.png)
![image](https://user-images.githubusercontent.com/119670575/205294585-35655179-23f4-41fc-806c-28125d8305fa.png)

**Describe the solution you'd like**
For list box we have come up with the design based on CDK components.
![image](https://user-images.githubusercontent.com/119670575/205294356-28db3ef8-76c1-4e3d-ade6-b9bb172ab188.png)

**Comment by Sr44102** [comment]
Any update on this component development? As this component is used very often in our project and this doesn't have any alternative components.

**Comment by kukoc** [comment]
Hi @Sr44102 

Halstack Designers will discuss and propose an alternative Design to this use case.

**Comment by kukoc** [comment]
More inputs shared by @Sr44102 

<p class="MsoNormal" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif; caret-color: rgb(33, 33, 33); color: rgb(33, 33, 33); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;"><span style="font-size: 11pt; line-height: 1.2;">Regarding the last pointer about styling the Typo:<o:p></o:p></span></p><ul type="disc" style="margin-bottom: 0cm; caret-color: rgb(33, 33, 33); color: rgb(33, 33, 33); font-family: Calibri; font-size: 16px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; margin-top: 0cm;"><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">Case bases we need to change if settings in the design, it will be a huge constraint to just work with the default settings for fonts.<span class="Apple-converted-space"> </span><o:p></o:p></span></li><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">I agree if we don’t add styling the code will be simple, but if we can have a option to create a style and that can be used across the project that would have been great.<o:p></o:p></span></li><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">For example I want the font with bold, Purple, 14px.<o:p></o:p></span></li></ul>

<DxcTypography                 fontWeight="400" color="#5F249F"fontSize="0.875rem"  letterSpacing="0.025em" >Place your text here</DxcTypography> | <DxcTypography class=”SmlPurpFont”>Place your text here</DxcTypography>
-- | --


<p class="MsoNormal" style="margin: 0cm 0cm 0cm 54pt; font-size: 10pt; font-family: Calibri, sans-serif; caret-color: rgb(33, 33, 33); color: rgb(33, 33, 33); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;"><span style="font-size: 11pt; line-height: 1.2;"><o:p> </o:p></span></p><ul type="disc" style="margin-bottom: 0cm; caret-color: rgb(33, 33, 33); color: rgb(33, 33, 33); font-family: Calibri; font-size: 16px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; margin-top: 0cm;"><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">If I can create a style and that could be added for required Typography tag, that helps a lot in<span class="Apple-converted-space"> </span><o:p></o:p></span></li><ul type="circle" style="margin-bottom: 0cm; margin-top: 0cm;"><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">Keeping the code clear<o:p></o:p></span></li><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">One place declaration<o:p></o:p></span></li><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">Any modifications will be easy<o:p></o:p></span></li><li class="MsoListParagraph" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif;"><span style="font-size: 11pt; line-height: 1.2;">Style consistence across the project<o:p></o:p></span></li></ul></ul><p class="MsoNormal" style="margin: 0cm; font-size: 10pt; font-family: Calibri, sans-serif; caret-color: rgb(33, 33, 33); color: rgb(33, 33, 33); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;"><span style="font-size: 11pt; line-height: 1.2;"><o:p> </o:p></span></p>

**Comment by kukoc** [comment]
Hi @Sr44102 , can you please share a link to your design so we can review it? 

![image](https://github.com/dxc-technology/halstack-react/assets/7778817/d2527df0-deb3-4a63-925c-408940fc51b6)

**Comment by Sr44102** [comment]
Hi @kukoc 
The Figma link for the multi selector design : [Multi Selector](https://www.figma.com/proto/MNMkI2t5G2mRIkPnBW2g8r/(BP)-Ceded-Portal?page-id=0%3A1&type=design&node-id=2706-24518&viewport=-23338%2C14406%2C0.51&t=SmYv3IjtiQu6lPEJ-1&scaling=min-zoom&starting-point-node-id=2101%3A17263&show-proto-sidebar=1&mode=design)
It's not fully functional.

**Comment by lcordobapuerto** [comment]
Hello @Sr44102 ,

We've reviewed the Figma file provided and have some comments regarding the design.

For the left container, we’d like to highlight some things:

1. We suggest using the scrollbar we currently have in our [Dropdown list component](https://developer.dxc.com/halstack/11/components/dropdown/)
![scrollbar](https://github.com/dxc-technology/halstack-react/assets/147498165/d67e0c69-34be-4d90-bb50-24e734f0db2c)

2. The icons to expand and collapse each dropdown list should be an “expand more” when the list is collapsed (the arrow pointing downwards) and an “expand less” (arrow pointing upwards) when the list is expanded. These arrows are being used in our [Accordion](https://developer.dxc.com/halstack/11/components/accordion/specifications/#specifications) component, please check this site for further reference.

In addition, regarding the buttons placed above these two containers, we suggest keeping just one primary button at the time depending on where the user is transferring from (left-to-right or right-to-left). This means:
- When the user is selecting items on the left, the button that carries the action to move said items to the right should be primary, while the button placed on the left needs to be secondary.
- However, if the user is selecting options on the left side, this button should be primary, and the other one secondary.
In short, the button associated witht the action that the user is performing is the one that should be primary, as having two primary actions one next to the other can be confusing for the user.

Another option that you can explore is having this type of layout with buttons separating both containers, like this:
![Alternative option](https://github.com/dxc-technology/halstack-react/assets/147498165/25c3d0af-23bd-42a6-9cca-074d927fef0d)

I hope this can help your team, and don’t hesitate to reach out again if you have more questions.
Thank you!

**Comment by kukoc** [comment]
Hi @Sr44102 

Just to be clear, we haven't started designing any Halstack component for this use case. As iván mentioned, it's quite specific and we should review in detail why the UX can't be accomplished using Halstack components. If that's really the case, you can go ahead and move forward with the recommendations provided by Lara. 

Please let us know, so we can close this issue in case we have an agreement. Thank you!

**Comment by Sr44102** [comment]
Hi @kukoc 
I can make those changes to the design, but we need to implement the component on our own right? As we have this requirement in many screens. 
Thanks @lcordobapuerto for the inputs, will get in touch in case of any clarifications.

**Comment by kukoc** [comment]
Hi @Sr44102 . Since Cede is under a full UX redesign, does it make sense to keep this issue open ar should we close it?

**Comment by Sr44102** [comment]
Hi @kukoc 
Sure. As we are exploring new designs for Cede, we can close this requirement.
